File: patterns/c-patterns/pattern96.c
// Pattern 96: ...

// 1 2 3 4 * 4 3 2 1
// 1 2 3 * * * 3 2 1
// 1 2 * * * * * 2 1
// 1 * * * * * * * 1
// * * * * * * * * *

#include <stdio.h>

int main()
{
    int n = 5;
    int i, j;

    int x = n;

    for (i = 1; i <= n; i++)
    {
        for (j = 1; j < 2 * n; j++)
        {
            if (j >= x && j <= n + i - 1)
            {
                printf("* ");
            }
            else if (j <= n)
            {
                printf("%d ", j);
            }
            else
            {
                printf("%d ", 2 * n - j);
            }
        }
        x--;
        printf("\n");
    }
    return 0;
}

File: patterns/python-patterns/pattern96.py
# Pattern 96: ...

# 1 2 3 4 * 4 3 2 1
# 1 2 3 * * * 3 2 1
# 1 2 * * * * * 2 1
# 1 * * * * * * * 1
# * * * * * * * * *

n = 5

x = n

for i in range(1, n + 1):
    for j in range(1, 2 * n):
        if j >= x and j <= n + i - 1:
            print("* ", end="")
        elif j <= n:
            print(j, end=" ")
        else:
            print(2 * n - j, end=" ")
    x -= 1
    print()
